src/Storages/StorageMergeTree.h

```diff
@@ -145,7 +145,198 @@ class StorageMergeTree final : public MergeTreeData
+                    {
+                        mutations.emplace(version, result.first->second);
+                    }
+                }
+            }
+            return result;
+        }
+
+        // FIXME Как появляются новые партишны в мапе??
+        std::optional<MergeTreeMutationEntry> pop(Int64 version)
+        {
+            auto mutation_it = all_mutations.find(version);
+            if (mutation_it != all_mutations.end())
+            {
+                if (mutation_it->second.partition_id)
+                {
+                    mutations_by_partition[*mutation_it->second.partition_id].erase(version);
+                }
+                else
+                {
+                    for (auto & [_, mutations] : mutations_by_partition)
+                    {
+                        mutations.erase(version);
+                    }
+                }
+                std::optional<MergeTreeMutationEntry> result(std::move(mutation_it->second));
+                all_mutations.erase(mutation_it);
+                return result;
+            }
+            else
+            {
+                return {};
+            }
+        }
+
+        auto erase(const MutationByVersionMap::iterator & mutation_it)
+        {
+            if (mutation_it->second.partition_id)
+            {
+                mutations_by_partition[*mutation_it->second.partition_id].erase(mutation_it->first);
+            }
+            else
+            {
+                for (auto & [_, mutations] : mutations_by_partition)
+                {
+                    mutations.erase(mutation_it->first);
+                }
+            }
+            return all_mutations.erase(mutation_it);
+        }
+
+        auto find(Int64 version) const
+        {
+            return all_mutations.find(version);
+        }
+
+        auto find(Int64 version)
+        {
+            return all_mutations.find(version);
+        }
+
+        bool empty() const
+        {
+            return all_mutations.empty();
+        }
+
+        size_t size() const
+        {
+            return all_mutations.size();
+        }
+
+        UInt64 lastVersion() const
+        {
+            return all_mutations.rbegin()->first;
+        }
+
+        auto begin() const
+        {
+            return all_mutations.begin();
+        }
+
+        auto lower_bound(UInt64 version) const
+        {
+            return all_mutations.lower_bound(version);
+        }
+
+        auto upper_bound(UInt64 version) const
+        {
+            return all_mutations.upper_bound(version);
+        }
+
+        auto end() const
+        {
+            return all_mutations.end();
+        }
+
+        auto begin()
+        {
+            return all_mutations.begin();
+        }
+
+        auto lower_bound(UInt64 version)
+        {
+            return all_mutations.lower_bound(version);
+        }
+
+        auto upper_bound(UInt64 version)
+        {
+            return all_mutations.upper_bound(version);
```

Comment:
hm... shouldn't it take one more arg (partition_id)

Reply:
This one was for all partitions.